Code/src/main/java/nl/utwente/group10/haskell/type/FuncT.java

```diff
@@ -30,4 +30,10 @@ public final String toHaskellType() {
         out.append(")");
         return out.toString();
     }
+
+    @Override
+    public FuncT getFresh() {
+        Type[] args = getFreshArgs();
+        return new FuncT(args[0], args[1]);
+    }
```

Comment:
I think that IF we are going to do this, it belongs in a new PR. But I think there are better ways to spend time now.